akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala

```diff
@@ -267,11 +267,8 @@ sealed trait TraversalBuilder {
   protected def internalSetAttributes(attributes: Attributes): TraversalBuilder
 
   def setAttributes(attributes: Attributes): TraversalBuilder = {
-    if (attributes ne Attributes.none) {
-      if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
-      else internalSetAttributes(attributes)
-    } else
-      this
+    if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
```

Comment:
Ah yes, you are right, this wanted to be an optimization to not do anything on Attributes.none, but this is handled elsewhere.

There is one worry now, that if you add a Dispatcher, then immediately set attributes to `none` then the async boundary remains.